packages/ra-ui-materialui/src/input/ArrayInput/ArrayInput.tsx

```diff
@@ -141,6 +147,21 @@ export const ArrayInput = (props: ArrayInputProps) => {
 
     const { error } = getFieldState(finalSource, formState);
 
+    const sourceContext = React.useMemo<SourceContextValue>(
+        () => ({
+            getSource: (source: string) => {
+                return source ? `${finalSource}.${source}` : finalSource;
```

Comment:
This I don't understand. In other instances, we check if there is a parent source context, then call that parent's getSource, otherwise return the source. Here you're doing the exact opposite...